### PR TITLE
refactor: factorize some code

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: zone.js [hunk 1]
```diff
@@ -114,38 +114,25 @@ Zone.patchSetClearFn = function (obj, fnNames) {
...
 
     if (delegate) {
+      var clearName = 'clear' + name;
       var ids = {};
 
-      if (setName === 'setInterval') {
-        zone[setName] = function (fn) {
-          var id;
-          arguments[0] = function () {
-            delete ids[id];
```
Comment (vicb) [inline zone.js]: Should we really `delete ids[id];` here ?

By deleting, `clearInterval()` will not call the `dequeueTask()` hook if called after the callback as triggered at least once, not sure if it makes sense ?

Comment (btford) [inline zone.js]: Not sure. Let's write a test to cover this case.

Comment (vicb) [inline zone.js]: ... and the expected behavior should be `dequeueTask()` always called (whenever clear is called), right ?

Comment (btford) [inline zone.js]: yes

**Comment by btford** [comment]
Landed as a3a11e90928c01336ca8035225cc52469e4b3f79.

**Pull request closed without merging**